contributors/design-proposals/componentconfig-dump.md

```diff
@@ -0,0 +1,164 @@
+
+## Design
+
+Components will have either a path indicated by annotations that can point to how to retrieve the configurations or through a well known component api path, as discussed below.
+
+### Component API Changes
+
+Components (like kubelet) have endpoints that can be altered to include an endpoint such as
+
+```
+ /configs 
+```
+
+similar to /metrics endpoints for K8s components and likely to be authenticated in much the same way.
+
+By way of an `HTTP GET` the output would be:
+
+```
+{
+    "kind": "configuration-list",
+    "metadata": {
+        "component": "kubelet",
+        "id": "818B908B-D053-CB11-BC8B-EEA826EBA090",
+        "timeStamp": "2016-09-13T14:47:35+00:00",
+    },
+    "configurations": [
+        {
+            "type": "componentconfigs",
+            "metadata": {
+                "version": "v1",
+                ...
+            },
+            "list": [
+                cloud-provider": "aws",
+                "aGroupName": {
+                    "aKey": "aValue",
+                    ...
+                },
+                ...
+            ]        
+        },
+        {
+            "type": "componentconfigs",
+            "metadata": {
+                "version": "v2",
+                ...
+            },
+            "list": [
+                cloud-provider": "aws",
+                "bGroupName": {
+                    "bKey": "bValue",
+                    ...
+                },
+                ...
+            ]        
+        },
+        {
+            "type": "additionalconfigs",
+            "list": [
+                "bKey": "bValue",
+                ...
+            ]
+        }
+    ]
+}
+```
+
+Those componentconfigs that are to be associated with a particular component will be annotated so as to be labeled for collection and served by the proposed endpoint.
+
+## Alternatives and Further Discussions
+
+Here we mention points of contentions and alternatives as we make progress in the proposal.
+
+* Security/Auth has been brought up as a point that strongly affects the proposed endpoint. Ideas around component endpoints and auth have been discussed for `/metrics` and it has been noted that SIG-Auth may be assigned the `/configs` endpoint to ensure auth is handled correctly.
+* The addition of an api change/addition to APIServer has been discussed, and for reasons of security it was agreed to be left out as a potential addition in the future.
+* The addition of kubectl/kubeadmin interactions has also been scrapped for reasons of security with possible addition in the future, but do not gain us much for our present motivations.
+* It has been agreed that extracting configs from componentconfigs is the way forward.
```

Comment:
I agree. Config reporting & setting is something everyone needs, not just Kubernetes components. We already have two types in the system for dealing with configs. If they are not sufficient for our own needs, they probably don't work for other users, either. We should fix them instead of building something that's special-purpose.